acid/features/history/model.py

```diff
@@ -59,34 +59,39 @@ def build_number(self):
         return None
 
     @classmethod
-    def get_branches(cls):
-        branches = select(b.ref for b in cls)
+    def get_all_branches(cls, pipeline):
+        branches = select(b.ref for b in cls if b.pipeline == pipeline)
         return branches
 
+    @classmethod
+    def get_latest_branches(cls, pipeline, max_branch_filter_size):
+        query = select((b.id, b.ref) for b in cls if b.pipeline == pipeline)
+        query = query.order_by(desc(1)).limit(max_branch_filter_size)
+        return {b[1] for b in query}
```

Comment:
I want to ask, what exactly do this method. Is `desc(1)` mean, that this is sorted by `b.ref` or `b.id`? 

Please, add unittest, it's new logic.